compiler/ExtractBase.ml

```diff
@@ -279,17 +294,24 @@ type formatter = {
 type id =
   | GlobalId of A.GlobalDeclId.id
   | FunId of fun_id
+  | DeclaredId of fun_id
```

Comment:
We shouldn't need this: declared ids should be of kind `FunId`

Reply:
Yeah my problem was that I needed a name for the definition, and a name for the usage (prefixed with `OpaqueDefs`), hence the two kinds of entries. If you know how to fix this cleanly, please go ahead? I'm a little confused by the handling of names right now

Reply:
Following a 1:1 discussion : we will remove `DeclaredId` and register the name with the prefix `OpaqueDefs.`

Reply:
Finally opted for the following:
- we register the definition names without any prefix, but remember which definitions are opaque
- at lookup time, depending on the usage, we add or not an "opaque_defs." prefix to the opaque definition names